Minor fix to AuthServiceProvider example

jamesggordon opened the pull request:
Addresses the issue raised here: http://stackoverflow.com/questions/35667780/lumen-authentication

taylorotwell commented:
I don't understand why people keep wanting to change this. The `api` driver is clearly defined in `AuthServiceProvider` and the default config in the framework repo points to `api`. What am I missing?

jamesggordon commented:
I’m new to Lumen so not really sure how everything ties together. What I can tell you is that changing ‘api’ to ‘token’ in the AuthServiceProvider class magically makes things work for a simple, out-of-the-box Lumen installation, which I guess is why people keep asking for it to be changed.

To be sure, I used “lumen new sandbox” to create a fresh installation of Lumen and then tried the following steps:

1. Used a browser to hit the default route - got the Lumen version message, as expected.

2. Uncommented the following two chunks of code in bootstrap/app.php to register the default auth middleware:

$app->routeMiddleware([
    'auth' => App\Http\Middleware\Authenticate::class,
]);

$app->register(App\Providers\AuthServiceProvider::class);

3. Modified app/Http/routes.php to add the auth middleware to the route:

$app->get('/',['middleware' => 'auth', function () use ($app) {
    return $app->version();
}]);

4. Hit the default route and got the following message:

lumen.ERROR: exception 'ErrorException' with message 'Undefined index: provider' in /Users/james/projects/sandbox/vendor/illuminate/auth/AuthManager.php:152

5. Made that one change to line 33 in App\Providers\AuthServiceProvider.php (changing ‘api’ to ’token’):

$this->app['auth']->viaRequest('token', function ($request) {

6. Hit the default route and got “Unauthorized”, which is what we expect.

I have since enabled Eloquent, implemented the api_token attribute in my User class, etc, and authentication is working as expected.


> On 11 Mar 2016, at 2:46 AM, Taylor Otwell <notifications@github.com> wrote:
> 
> I don't understand why people keep wanting to change this. The api driver is clearly defined in AuthServiceProvider and the default config in the framework repo points to api. What am I missing?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/laravel/lumen/pull/59#issuecomment-194914834>.
>
